client/src/components/header/index.tsx

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import Nav from "../nav";
+import { Container } from "../../box/absoluteBox";
+import { ButtonSecondary, HeaderSection, LogoTitle, LogoSubtitle,  HeaderWrapper, Button, FlipInX } from "./header";
+import {Link} from 'react-router-dom'
+
+const Header:React.FC = () => {

```

Comment:
Try to avoid usage of React.FC

Reply:
I'll correct it on React.ReactNode

Reply:
I can't use React.ReactNode for Header or Nav component because type of ReactNode is not valid for JSX.

Reply:
how can i label a component without using React.FC?

Reply:
Review this answer https://stackoverflow.com/questions/58123398/when-to-use-jsx-element-vs-reactnode-vs-reactelement#:~:text=Element%20is%20ReactElement%20%2C%20whose%20props,render()%20in%20class%20components.

Reply:
Try JSX.Element

Reply:
const MyComp2 = (): JSX.Element => <div>Hello</div>; 
const MyComp3 = (): React.ReactElement => <div>Hello</div>;  
// Option 3 is equivalent to 2 + we don't need to use a global (JSX namespace)

// Use built-in `FunctionComponent` or `FC` type
const MyComp4: React.FC<MyProps> = () => <div>Hello</div>;

Note: Avoiding React.FC won't save you from the JSX.Element | null return type restriction.

Create React App recently dropped React.FC from its template, as it has some quirks like an implicit {children?: ReactNode} type definition. **So using React.FC sparingly might be preferable.**

In edge cases, you can add a type assertion or Fragments as workaround: